godtools/App/Services/Renderer/Analytics/MobileContentAnalytics.swift

```diff
@@ -23,27 +24,28 @@ class MobileContentAnalytics {
         self.analyticsSystems = analyticsSystems
     }
         
-    func trackEvents(events: [AnalyticsEventModelType], rendererPageModel: MobileContentRendererPageModel) {
+    func trackEvents(events: [AnalyticsEvent], renderedPageContext: MobileContentRenderedPageContext) {
         
         for event in events {
-            trackEvent(event: event, rendererPageModel: rendererPageModel)
+            trackEvent(event: event, renderedPageContext: renderedPageContext)
         }
     }
     
-    private func trackEvent(event: AnalyticsEventModelType, rendererPageModel: MobileContentRendererPageModel) {
+    private func trackEvent(event: AnalyticsEvent, renderedPageContext: MobileContentRenderedPageContext) {
         
         guard let action = event.action, !action.isEmpty else {
             return
         }
                 
-        let data: [String: String] = event.getAttributes()
+        let data: [String: String] = event.attributes
+        let systems: [AnalyticsEvent.System] = Array(event.systems)
         
-        for system in event.systems {
+        for system in systems {
              
-            if let analyticsSystem = analyticsSystems[system.lowercased()] {
+            if let analyticsSystem = analyticsSystems[system.name.lowercased()] {
```

Comment:
maybe an enhancement here is to define the `analyticsSystems` dictionary to use the System enum values as the keys instead of having to convert them to strings.

Reply:
I'm not sure if this is actually possible, the documentation says the dictionary key needs to be hashable. I'm not sure if a Kotlin enum implements that protocol or not